concept/validation/migration.feature

```diff
@@ -0,0 +1,153 @@
+#
+# Copyright (C) 2022 Vaticle
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+#noinspection CucumberUndefinedStep
+Feature: Data validation
+
+  Background:
+    Given typedb starts
+    Given connection opens with default authentication
+    Given connection has been opened
+    Given connection does not have any database
+    Given connection create database: typedb
+    Given connection open schema session for database: typedb
+    Given session opens transaction of type: write
+
+
+  Scenario: An ownership can be moved down one type, with data in place at the lower levels
```

Comment:
I feel like there are more 'user' operations that we should test if we've got these. Let's cross check w/ james?

Some ideas:

- Moving a type to a different super type, with existing owns, plays, and data in place (all in one maybe?)
  - Variation on this that might require a multi-step operation to adjust the data?
- Moving a capability up/down the hierarchy (you have the downward ones) with data in place